web/src/components/EditTest/EditTest.tsx

```diff
@@ -37,8 +41,8 @@ const EditTest = ({test}: IProps) => {
           </Button>
           <Button
             data-cy="edit-test-submit"
-            loading={isEditLoading}
-            disabled={!isValid}
+            loading={!stateIsFinished || isEditLoading}
```

Comment:
After the save button is done the new version of the test should be available, for me the loading state has to match the form save request, not the trace process. I think the way it was matched that.

Reply:
I have done this without changing the loading prop. But the button flicks the loading very fast because the api responds very fast. It looks weird. And second, because the button label reads “Save & Run” so it makes sense for me having it loading while it is running. But it is pretty easy to just revert that line. What should we do? Maybe @olha23 can chime in. I will record a loom video showing the difference

Reply:
I have reverted back the loading prop to what it was so we can approve this.
We can discuss this behaviour on the parking lot tommorow @xoscar